solid-terms.ttl

```diff
@@ -1,212 +1,270 @@
-@base <http://www.w3.org/ns/solid/terms> .
-@prefix : <#> .
-@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
-@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
-@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
-@prefix owl: <http://www.w3.org/2002/07/owl#> .
-@prefix cc: <https://creativecommons.org/ns#> .
-@prefix dcterms: <http://purl.org/dc/terms/> .
-@prefix log: <http://www.w3.org/2000/10/swap/log#> .
-@prefix ldp: <http://www.w3.org/ns/ldp#> .
-
-<>
+prefix solid: <http://www.w3.org/ns/solid/terms#>
+prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+prefix owl: <http://www.w3.org/2002/07/owl#>
+prefix xsd: <http://www.w3.org/2001/XMLSchema#>
+prefix dc: <http://purl.org/dc/terms/>
+prefix ldp: <http://www.w3.org/ns/ldp#>
+prefix log: <http://www.w3.org/2000/10/swap/log#>
+prefix vann: <http://purl.org/vocab/vann/>
+
+<http://www.w3.org/ns/solid/terms>
     a owl:Ontology ;
-    cc:attributionURL <> ;
-    cc:license <https://creativecommons.org/publicdomain/zero/1.0/> ;
-    dcterms:issued "2015-11-16"^^xsd:date ;
-    dcterms:modified "2018-01-24"^^xsd:date ;
-    rdfs:label "Solid terms"@en .
+    dc:license <https://creativecommons.org/publicdomain/zero/1.0/> ;
+    rdfs:label "Solid Terms ontology"@en ;
+    rdfs:comment "Solid, derived from social linked data, is a proposed set of conventions and tools for building decentralized social applications based on Linked Data principles."@en ;
+    rdfs:isDefinedBy <http://www.w3.org/ns/solid/terms> ;
+    dc:issued "2015-11-16"^^xsd:date ;
+    dc:modified "2015-12-18"^^xsd:date ;
+    dc:modified "2015-12-20"^^xsd:date ;
+    dc:modified "2016-01-04"^^xsd:date ;
+    dc:modified "2016-02-05"^^xsd:date ;
+    dc:modified "2016-03-11"^^xsd:date ;
+    dc:modified "2016-05-17"^^xsd:date ;
+    dc:modified "2017-08-15"^^xsd:date ;
+    dc:modified "2017-08-16"^^xsd:date ;
+    dc:modified "2017-08-17"^^xsd:date ;
+    dc:modified "2018-01-24"^^xsd:date ;
+    dc:modified "2018-10-24"^^xsd:date ;
+    dc:modified "2018-10-25"^^xsd:date ;
+    dc:modified "2018-10-26"^^xsd:date ;
+    dc:modified "2018-11-07"^^xsd:date ;
+    dc:modified "2019-01-22"^^xsd:date ;
```

Comment:
```suggestion
```

Okay to keep only one last modified. Each term in the vocab will have its own issued/modified any way. Don't need to bloat this.

Reply:
Excellent suggestion, I dislike redundant information. What about actually keeping only the dc:modified that correspond to the dates where the term was modified. I think adding all modified dates was sloppy. I don't think keeping only one/the last modified date is coherent.

Reply:
Keeping one `dc:modified ` at the file level is definitely useful. It's trivial "bloat". I would do this.

Keeping all `dc:modified` might be useful, at either the file level or the term level. I don't have a specific use case, but someone else might, now or in the future. I don't have a strong argument in favor of keeping all, but it also seems a small "bloat" and doesn't feel especially redundant to me.

Removing all `dc:modified` seems like overkill, at any level.

Reply:
But the ontology URI is not the file level.

I'm not sure I'm expressing myself clearly. I think the ontology just like all the classes and properties it defines should use the `dc:modified` predicate to denote the dates at which it has been modified (for example if the label or comment associated with it has been ammended).

Keeping the last modified as the ontology's only `dc:modified` property is misleading and removes information about when the ontology itself was modified. I think the rule for using dc:modified should be consistently applied to the ontology and the things it defines. If one wants to get the latest date at which any term defined by the ontology has been modified, it's easy to do:

```sparql
SELECT ?lastModified {
    ?s <http://purl.org/dc/terms/modified> ?lastModified .
}
ORDER BY DESC(?lastModified)
LIMIT 1
```

And you can easily isolate the ontology in a larger graph thanks to the rdfs:isDefinedBy property:

```sparql
SELECT ?s {
    ?s rdfs:isDefinedBy solid:
}
```

As to the use case, one might find useful to know when things change and one might not want to lose data. Therefore, a more recent dc:modified should not overwrite an earlier one. It would be trivial to regularly pool a namespace looking for modifications. Ideally, you could easily identify edits through this.

Reply:
@matthieubosquet - You seem now to be arguing against your earlier position... Then to remove, and now to keep, all the `dc:modified`, across the board.

(But they still show as being removed by this PR.)

Having considered things a bit more, I think there is indeed value in retaining the  multiple dates of changes to the ontology, to each term therein, and to the document where the ontology is serialized.  It shows that attention is paid to the ontology on a regular basis, and that error corrections and refinements are applied as understanding deepens.

All of which is to say -- let's keep all the `dc:modified`.

Reply:
To be clear, I meant that a single dcterms:modified for http://www.w3.org/ns/solid/terms will suffice. The value can be updated whenever a change happens.

Each property and class in the vocab can have its own issued and modified as well. That will sufficel.

I have no strong objection to having multiple dcterms:modified values per subject but don't see high value at this time - granted this is a very subjective view. It goes without saying that provenance record is always important. So, if that's really the focus here, I suggest to consider including other information eg. comment/reason for the update. Many ways to tackle this.. eg. in practice some vocabs/ontologies use a single rdfs:comment to include the changes whenever they happen with a date field. Could even take it further and use PROV-O..

Reply:
Ammended and:
1. Kept only the latest dc:modified on the ontology
2. Kept the all dc:modified on the properties and classes

Provenance-wise, I think it would be good to leverage what happens in version control systems. Mainly because it is in my opinion the best authoritative source of information about collaborative development of ontologies.

That is, for example, linking to the previous version of the ontology via the raw commit URI and pull request.

Here we could for example:
```turtle
<http://www.w3.org/ns/solid/terms#>
    a owl:Ontology ;
    dc:license <https://creativecommons.org/publicdomain/zero/1.0/> ;
    rdfs:label "Solid Terms"@en ;
    rdfs:comment "The Solid Terms vocabulary defines terms used in the Solid API specifications."@en ;
    rdfs:isDefinedBy <http://www.w3.org/ns/solid/terms#> ;
    dc:issued "2015-11-16"^^xsd:date ;
    dc:modified "2020-12-22"^^xsd:date ;
    owl:versionInfo [
            dc:date "2020-12-22"^^xsd:date ;
            dc:replaces <https://raw.githubusercontent.com/solid/vocab/387df145ef39936120b25b48891315fe53cb4d91/solid-terms.ttl> ;
            rdfs:seeAlso <https://github.com/solid/vocab/pull/46> ;
        ] ;
    vann:preferredNamespacePrefix "solid" ;
    vann:preferredNamespaceUri "http://www.w3.org/ns/solid/terms#"^^xsd:anyURI .
```

The reason I kept the dc:modified on the properties and classes is: it makes it easy to track when and how a term was changed via the GitHub commits. A luxury that is not afforded easily trying to go through changes to a single file.

Reply:
Actually pushed the proposed approach and removed the dc:modified which is inconsistent with the rest of the document.